src/index.ts

```diff
@@ -1,61 +1,111 @@
-  _CONNECTOR_NAME_ConnectorEventOptions
+function validateBaseURL(url?: string): string {
+  if (typeof url !== 'string') {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  const match = url.match(/^(https:\/\/[\w-]+(\.[\w-]+)*(:\d{1,5})?)\/?$/)
+  if (!match) {
+    throw new Error(`Invalid url: ${url}`)
+  }
+  return match[1]
+}
+
+export default class ShopifyConnector extends BaseConnector<
+  ShopifyConnectorConfigOptions,
+  ShopifyConnectorEventOptions
 > {
-  // Your class variables
-  var1: string
+  _sdk: Shopify
 
-  constructor(app: Reshuffle, options?: _CONNECTOR_NAME_ConnectorConfigOptions, id?: string) {
+  constructor(app: Reshuffle, options: ShopifyConnectorConfigOptions, id?: string) {
+    const { baseURL, webhookPath, webhookName, ...shopifyOptions } = options
     super(app, options, id)
-    this.var1 = options?.var1 || 'initial value'
-    // ...
+    this._sdk = new Shopify(shopifyOptions)
   }
 
-  onStart(): void {
-    // If you need to do something specific on start, otherwise remove this function
-  }
+  async onStart(): Promise<void> {
+    const logger = this.app.getLogger()
+    const events = Object.values(this.eventConfigurations)
+    if (events.length) {
+      const url = validateBaseURL(this.configOptions?.baseURL)
+      const address = url + (this.configOptions?.webhookPath || DEFAULT_WEBHOOK_PATH)
+
+      const webhooks = await this._sdk.webhook.list()
 
-  onStop(): void {
-    // If you need to do something specific on stop, otherwise remove this function
+      for (const event of events) {
+        const {
+          options: { topic },
+        } = event
+        // Check existing webhook
+        const existingWebhook = webhooks.find(
+          (opts) => opts.address === address && opts.topic === topic,
+        )
+
+        const webhook = existingWebhook || (await this._sdk.webhook.create({ address, topic }))
+
+        if (webhook.created_at) {
+          logger.info(
+            `Reshuffle Shopify - webhook registered successfully (topic: ${webhook.topic}, address ${webhook.address})`,
+          )
+        } else {
+          logger.error(
+            `Reshuffle Shopify - webhook registration failure (topic: ${webhook.topic}, address: ${webhook.address})`,
+          )
+        }
+      }
+    }
   }
 
-  // Your events
-  on(
-    options: _CONNECTOR_NAME_ConnectorEventOptions,
-    handler: any,
-    eventId: string,
-  ): EventConfiguration {
+  on(options: ShopifyConnectorEventOptions, handler: any, eventId: string): EventConfiguration {
+    const path = this.configOptions?.webhookPath || DEFAULT_WEBHOOK_PATH
+
     if (!eventId) {
-      eventId = `_CONNECTOR_NAME_/${options.option1}/${this.id}`
+      eventId = `Shopify${path}/${options.topic}/${this.id}`
     }
     const event = new EventConfiguration(eventId, this, options)
     this.eventConfigurations[event.id] = event
 
     this.app.when(event, handler)
+    this.app.registerHTTPDelegate(path, this)
```

Comment:
Does this need to be in `on` or `onStart` ?

Reply:
I think it's make more sense in the `on` as the `onStart` happens just after app.start() which means the server is already created. I'm surprised that it works for Monday connector...